core/src/main/java/nl/nn/adapterframework/align/XmlTypeToJsonSchemaConverter.java

```diff
@@ -145,14 +146,40 @@ public JsonStructure getDefinition(XSTypeDefinition typeDefinition) {
 			return builder.build();
 		} else {
 			XSSimpleTypeDefinition simpleTypeDefinition = (XSSimpleTypeDefinition)typeDefinition;
-			if (DEBUG) log.debug("typeDefinition.name ["+typeDefinition.getName()+"]");
+			if (DEBUG) log.debug("typeDefinition.name ["+typeDefinition.getName()+"]");;
+			if (DEBUG) log.debug("simpleTypeDefinition.getBuiltInKind ["+simpleTypeDefinition.getBuiltInKind()+"]");
 			if (DEBUG) log.debug(ToStringBuilder.reflectionToString(typeDefinition,ToStringStyle.MULTI_LINE_STYLE));
+
  			JsonObjectBuilder builder = Json.createObjectBuilder();
-			if (simpleTypeDefinition.getNumeric()) {
-				builder.add("type", "number");
-			} else if (simpleTypeDefinition.getBuiltInKind() == XSConstants.BOOLEAN_DT) {
+			short builtInKind = simpleTypeDefinition.getBuiltInKind();
+			String dataType = getJsonDataType(builtInKind);
+			
+			if (dataType.equalsIgnoreCase("integer") || dataType.equalsIgnoreCase("number")) {
+				builder.add("type", dataType.toLowerCase());
+
+				applyFacet(simpleTypeDefinition, builder, "maximum", XSSimpleTypeDefinition.FACET_MAXINCLUSIVE);
+				applyFacet(simpleTypeDefinition, builder, "minimum", XSSimpleTypeDefinition.FACET_MININCLUSIVE);
+				applyFacet(simpleTypeDefinition, builder, "exclusiveMaximum", XSSimpleTypeDefinition.FACET_MAXEXCLUSIVE);
+				applyFacet(simpleTypeDefinition, builder, "exclusiveMinimum", XSSimpleTypeDefinition.FACET_MINEXCLUSIVE);
+				applyFacet(simpleTypeDefinition, builder, "enumeration", XSSimpleTypeDefinition.FACET_ENUMERATION);
+			} else if (dataType.equalsIgnoreCase("boolean")) {
 				builder.add("type", "boolean");
+			} else if (dataType.equalsIgnoreCase("string")) {	
+				builder.add("type", "string");
+			
+				applyFacet(simpleTypeDefinition, builder, "maxLength", XSSimpleTypeDefinition.FACET_MAXLENGTH);
+				applyFacet(simpleTypeDefinition, builder, "minLength", XSSimpleTypeDefinition.FACET_MINLENGTH);
+				applyFacet(simpleTypeDefinition, builder, "pattern", XSSimpleTypeDefinition.FACET_PATTERN);
+				applyFacet(simpleTypeDefinition, builder, "enumeration", XSSimpleTypeDefinition.FACET_ENUMERATION);
+			} else if (dataType.equalsIgnoreCase("date") || dataType.equalsIgnoreCase("date-time") || dataType.equalsIgnoreCase("time")) {		
+				builder.add("type", "string");
+				
+				builder.add("format", dataType);
+
+				applyFacet(simpleTypeDefinition, builder, "pattern", XSSimpleTypeDefinition.FACET_PATTERN);
+				applyFacet(simpleTypeDefinition, builder, "enumeration", XSSimpleTypeDefinition.FACET_ENUMERATION);
```

Comment:
Zou je nog een test kunnen maken (of Datatypes uitbreiden) voor 'pattern' en 'enumeration'? Die kom ik niet tegen in de test resultaten, en ik zie de applyFacet() ervoor hierboven twee keer.

Reply:
Ik zie het, ik ben even gaan kijken en het klopt. 

In eerste instantie werden alleen single valued facets behandeld. Ik heb de method applyFacet() aangepast om ook multi valued facets aan te kunnen.

In de aanwezige tests zijn inmiddels dus ook tests cases voor pattern en enumeration.